units/src/amount/signed.rs

```diff
@@ -63,7 +63,34 @@ impl SignedAmount {
     pub const MAX: Self = SignedAmount::MAX_MONEY;
 
     /// Constructs a new [`SignedAmount`] with satoshi precision and the given number of satoshis.
-    pub const fn from_sat(satoshi: i64) -> SignedAmount { SignedAmount(satoshi) }
+    ///
+    /// # Errors
+    ///
+    /// On values exceeding [`SignedAmount::MAX`] or less than [`SignedAmount::MIN`]
+    pub const fn from_sat(satoshi: i64) -> Result<SignedAmount, OutOfRangeError> {
+        if satoshi < Self::MIN.0 {
+            Err(OutOfRangeError { is_signed: true, is_greater_than_max: false })
+        } else if satoshi > Self::MAX.0 {
+            Err(OutOfRangeError { is_signed: true, is_greater_than_max: true })
+        } else {
+            Ok(SignedAmount(satoshi))
+        }
+    }
+
+    /// Constructs a new [`SignedAmount`] with satoshi precision and the given number of satoshis.
+    ///
+    /// # Panics
+    ///
+    /// On values exceeding [`SignedAmount::MAX`] or less than [`SignedAmount::MIN`]
+    pub const fn from_sat_unchecked(satoshi: i64) -> SignedAmount {
+        if satoshi < Self::MIN.0 {
+            panic!("Out Of Range Error: less than MIN")
+        } else if satoshi > Self::MAX.0 {
+            panic!("Out Of Range Error: greater than MAX")
+        } else {
+            SignedAmount(satoshi)
+        }
+    }
```

Comment:
I believe this is incorrect code for the function nomenclature. The function name includes `_unchecked` but the code checks the value. IMO this function should just ignore the invariant.